### PR TITLE
[wip] outline next round of tests

### PR DESCRIPTION
**Pull request by josh-works**
this is a quick example PR of a good way to have very focused conversations around the code. 

In the 'files changed' tab, you'll see comments I'm adding on specific things. You can leave comments here, and as I push up changes to this branch, they'll always be captured here.

### CODE REVIEW

File: test/binary_search_tree_test.rb [hunk 1]
```diff
@@ -88,4 +88,24 @@ def test_it_can_insert_a_twelfth_node
     assert_equal 3, result
   end
 
+  def test_include?
+    @tree.insert(61, "Bill & Ted's Excellent Adventure")
+
+    assert @tree.include?(61)
+  end
```
Comment (josh-works) [inline test/binary_search_tree_test.rb:135]: True TDD. Obv. the required methods don't exist yet, but here's some ideas.

These comments even have nice code formatting options:

```ruby
def test_it_can_insert_a_twelfth_node
    @tree.insert(61, "Bill & Ted's Excellent Adventure")
    
    assert stuff
  end
```

Finally, you can @Shniks  someone and they'll get a notification to take a look.

File: test/binary_search_tree_test.rb [hunk 2]
```diff
@@ -88,4 +88,24 @@ def test_it_can_insert_a_twelfth_node
...
+
+  def test_depth_of_single_node
+    @tree.insert(61, "Bill & Ted's Excellent Adventure")
+
+    assert_equal 1, @tree.depth_of(61)
+  end
+
+  def test_depth_of_several_nodes
+    @tree.insert(61, "Bill & Ted's Excellent Adventure")
```
Comment (josh-works) [inline test/binary_search_tree_test.rb:144]: all of the copying `@tree.insert(int, "title")` seems ripe for a `setup` method.

Comment (Shniks) [reply test/binary_search_tree_test.rb:144]: I created the setup method by using a hash with the score as the keys and the movie title and  its depth (both in an array), as the values. Then I use .each to iterate through the hash and do the insertions or assertions.

**Comment by Shniks** [comment]
I also created the tests and methods for max, min and depth_of

**Comment by josh-works** [comment]
OK, your code is fantastic. Well done on this project. 

I agree w/your decision to move on to SalesEngine. 

One last comment, and I put an example in the most recent commit, but I'd recommend keeping this in mind in SalesEngine - you can use a "test helper" file to handle all your `requires` statements, and to avoid copying test setup code. Take a look at the comments in `test_helper.rb` and in your `binary_search_tree_test.rb`

File: test/binary_search_tree_test.rb [hunk 3]
```diff
@@ -1,6 +1,5 @@
-require 'minitest/autorun'
-require 'minitest/pride'
-require './lib/binary_search_tree'
+require 'test_helper'
+
```
Comment (josh-works) [inline test/binary_search_tree_test.rb:3]: All of your test files now have a single `require 'test_helper'`, that calls... the test helper!

File: test/binary_search_tree_test.rb [hunk 4]
```diff
@@ -75,9 +74,7 @@ def test_it_can_find_depth_of_each_movie_when_inserting_twelve_movies
   end
 
   def test_it_can_verify_presence_of_score_in_the_tree
-    @movies_list.each do |key, value|
-      result = @tree.insert(key, value[0])
-    end
+    load_movies
```
Comment (josh-works) [inline test/binary_search_tree_test.rb:78]: I moved this method into the test file, exactly as is, and wrapped it in a `load_movies` method. Now you can call this from your tests, as needed. 

(You don't use it everywhere, so it couldn't be a setup method.)

File: test/test_helper.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+require 'minitest/autorun'
+require 'minitest/pride'
+require './lib/node'
+require './lib/binary_search_tree'
+
```
Comment (josh-works) [inline test/test_helper.rb:5]: for the test helper to work everywhere, you need to give it _every_ require you'll need in your project.

Obviously you can add something like `pry` here, and it'll be available in all of your tests.

**Pull request closed without merging**